source/validate.cpp

```diff
@@ -518,6 +436,109 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+// Allows for piping of multiple functions which return spv_result_t
+spv_result_t operator||(spv_result_t lhs, spv_result_t rhs) {
```

Comment:
I'm ok with having an overload, but the ordinary || operator is short-circuiting, but this implementation isn't.  The RHS will always be evaluated.   Better to make it "operator|".
With that, the precedence changes too, but if that ever matters, use parens to group the expression.